lib/worker/master.rb

```diff
@@ -0,0 +1,90 @@
+# frozen_string_literal: true
+module Worker
+  # The Master Worker is the entrypoint to trade capture. The single 'work!' method will block continuously.
+  class Master
+    include ::ErrorNotifierMethods
+    include ::Logging
+
+    def initialize
+      Logging.logger.info { 'Creating Worker::Master' }
+      @paused = false
+      @active_accounts = []
+    end
+
+    # work is a blocking method. The 'fetch' and 'pause' methods have custom timeouts and are implemented
+    # in each method.
+    def work!
+      loop do
+        log_activity
+        inside_operating_window? ? fetch : pause
+      end
+    end
+
+    private
+
+    def fetch
+      fetch_active_accounts!.each do |account_hash|
```

Comment:
This might be the best place to preserve threading. If one request ran long all of the others would wait on it. It might make sense to fire off all (or a few) of the requests in separate threads and then join them all and do the loop again.